src/MilkdropXBMC.cpp

```diff
@@ -118,59 +131,52 @@ void urlEscape(std::string& str)
   replaceAll(str, "#",  "%23");
 }
 
-extern "C" ADDON_STATUS ADDON_Create(void* hdl, void* props)
+//-- Create -------------------------------------------------------------------
+// Called on load. Addon should fully initalize or return error status
+// !!! Add-on master function !!!
+//-----------------------------------------------------------------------------
+ADDON_STATUS CVisualizationMilkdrop::Create()
 {
-  if (!props)
-    return ADDON_STATUS_UNKNOWN;
-
-	KODI = new ADDON::CHelper_libXBMC_addon;
-	if (!KODI->RegisterMe(hdl))
-	{
-		delete KODI;
-		KODI = NULL;
-		return ADDON_STATUS_PERMANENT_FAILURE;
-	}
-	
-	AddonProps_Visualization* visprops = (AddonProps_Visualization*)props;
-  _mkdir(visprops->profile);
+  _mkdir(Profile().c_str());
 
-  std::string presets = std::string(visprops->presets).append("\\presets\\");
+  std::string presets = Presets().append("\\presets\\");
   urlEscape(presets);
   g_presetsDir = "zip://" + presets;
 
-  Preinit();
-  if(!g_plugin || !g_plugin->PluginInitialize((ID3D11DeviceContext*)visprops->device, visprops->x, visprops->y, visprops->width, visprops->height, visprops->pixelRatio))
+  if (!g_plugin)
+  {
+    g_plugin = new CPlugin;
+    g_plugin->PluginPreInitialize(0, 0);
+  }
+
+  if (!g_plugin || !g_plugin->PluginInitialize(static_cast<ID3D11DeviceContext*>(Device()), X(), Y(), Width(), Height(), PixelRatio()))
     return ADDON_STATUS_UNKNOWN;
 
   return ADDON_STATUS_NEED_SAVEDSETTINGS; // We need some settings to be saved later before we quit this plugin
 }
 
-extern "C" void Start(int iChannels, int iSamplesPerSec, int iBitsPerSample, const char* szSongName)
-{}
+bool CVisualizationMilkdrop::Start(int iChannels, int iSamplesPerSec, int iBitsPerSample, std::string szSongName)
+{
+  return true;
+}
```

Comment:
If you not need it can it completely stay away. But can also be leaved if needed in future for improvement.